src/rai_core/rai/tools/ros2/nav2.py

```diff
@@ -0,0 +1,290 @@
+# Copyright (C) 2025 Robotec.AI
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import base64
+import time
+from typing import List, Optional, Type, cast
+
+import cv2
+import numpy as np
+from geometry_msgs.msg import Point, PoseStamped, Quaternion, TransformStamped
+from langchain_core.tools import BaseTool
+from nav2_msgs.action import NavigateToPose
+from nav_msgs.msg import OccupancyGrid
+from pydantic import BaseModel, Field
+from rclpy.action import ActionClient
+from tf_transformations import euler_from_quaternion, quaternion_from_euler
+
+from rai.communication.ros2 import ROS2ARIMessage
+from rai.communication.ros2.connectors import ROS2ARIConnector
+from rai.messages.multimodal import MultimodalArtifact
+from rai.tools.ros2.base import BaseROS2Tool, BaseROS2Toolkit
+
+action_client: Optional[ActionClient] = None
+current_action_id: Optional[str] = None
+current_feedback: Optional[NavigateToPose.Feedback] = None
+current_result: Optional[NavigateToPose.Result] = None
+
+
+class Nav2Toolkit(BaseROS2Toolkit):
+    connector: ROS2ARIConnector
+
+    def get_tools(self) -> List[BaseTool]:
+        return [
+            NavigateToPoseTool(connector=self.connector),
+            CancelNavigateToPoseTool(connector=self.connector),
+            GetNavigateToPoseFeedbackTool(connector=self.connector),
+            GetNavigateToPoseResultTool(connector=self.connector),
+        ]
+
+
+class NavigateToPoseToolInput(BaseModel):
+    x: float = Field(..., description="The x coordinate of the pose")
+    y: float = Field(..., description="The y coordinate of the pose")
+    z: float = Field(..., description="The z coordinate of the pose")
+    yaw: float = Field(..., description="The yaw angle of the pose")
+
+
+class NavigateToPoseTool(BaseROS2Tool):
+    name: str = "navigate_to_pose"
+    description: str = "Navigate to a specific pose"
+
+    args_schema: Type[NavigateToPoseToolInput] = NavigateToPoseToolInput
+
+    frame_id: str = "base_link"
```

Comment:
With the current configuration of [nav2 for rosbot](https://github.com/RobotecAI/rai-rosbot-xl-demo/blob/c71752c088425b73b26191ef2f55d2bdecf218f4/run-nav.bash#L1) `frame_id` has to be set to `map`
```suggestion
    frame_id: str = "map"
```